core/rpc/client/gateway/http/client.go

```diff
@@ -120,20 +120,16 @@ func (g *GatewayHttpClient) GetAuthParameter(ctx context.Context) (*types.Gatewa
 		return nil, err
 	}
 
-	if r.Error != "" {
-		return nil, errors.New(r.Error)
-	}
-
 	return r.Result, nil
 }
 
-// gatewayAuthPostResponse defines the response of POST request for /auth
-type gatewayAuthPostResponse struct {
-	Error string `json:"error"`
+// gatewayErrResponse defines the response of gateway error.
+type gatewayErrResponse struct {
+	// to be compatible with google.golang.org/genproto/googleapis/rpc/status.Status
+	Message string `json:"message"`
 }
 
 // gatewayAuthGetResponse defines the response of GET request for /auth
 type gatewayAuthGetResponse struct {
 	Result *types.GatewayAuthParameter `json:"result"`
-	Error  string                      `json:"error"`
```

Comment:
All KGW error will be in gatewayErrRespose, this is no use.

Reply:
For test on main, I guess not, because we don't really test the error for KGW APIs?

For test on #29, not sure